src/server/object_services/map_reader.js

```diff
@@ -104,59 +109,50 @@ async function read_parts_mapping(parts) {
 }
 
 
-// /**
-//  * @param {nb.Chunk[]} chunks
-//  * @param {nb.LocationInfo} [location_info]
-//  */
-// async function update_chunks_on_read(chunks, location_info) {
-//     const chunks = _.map(parts, 'chunk');
-//     const tiering_status_by_bucket_id = {};
-
-//     for (const chunk of chunks) {
-//         map_server.populate_chunk(chunk);
-//     }
-
-//     await _load_chunk_mappings(chunks, tiering_status_by_bucket_id);
-
-//     const chunks_to_scrub = [];
-//     try {
-//         const bucket = system_store.data.get_by_id(chunks[0].bucket);
-//         const tiering_status = tiering_status_by_bucket_id[bucket._id];
-//         const selected_tier = mapper.select_tier_for_write(bucket.tiering, tiering_status);
-//         for (const chunk of chunks) {
-//             map_server.populate_chunk(chunk);
-//             if (!chunk.tier._id || !_.isEqual(chunk.tier._id, selected_tier._id)) {
-//                 dbg.log0('Chunk with low tier will be sent for rebuilding', chunk);
-//                 chunks_to_scrub.push(chunk);
-//             } else if (location_info) {
-//                 const chunk_info = mapper.get_chunk_info(chunk);
-//                 const mapping = mapper.map_chunk(chunk_info, chunk.tier, bucket.tiering, tiering_status, location_info);
-//                 if (mapper.should_rebuild_chunk_to_local_mirror(mapping, location_info)) {
-//                     dbg.log2('Chunk with following mapping will be sent for rebuilding', chunk, mapping);
-//                     chunks_to_scrub.push(chunk);
-//                 }
-//             }
-//         }
-//         if (chunks_to_scrub.length) {
-//             dbg.log1('Chunks wasn\'t found in local pool - the following will be rebuilt:', util.inspect(chunks_to_scrub));
-//             await server_rpc.client.scrubber.build_chunks({
-//                 chunk_ids: _.map(chunks_to_scrub, '_id'),
-//                 tier: selected_tier._id,
-//             }, {
-//                 auth_token: auth_server.make_auth_token({
-//                     system_id: chunks_to_scrub[0].system,
-//                     role: 'admin'
-//                 })
-//             });
-//         }
-//     } catch (err) {
-//         dbg.warn('Chunks failed to rebuilt - skipping');
-//     }
-//     if (chunks_to_scrub.length) {
-//         // mismatch type...
-//         await MDStore.instance().load_blocks_for_chunks(chunks);
-//     }
-// }
+/**
+ * @param {nb.Chunk[]} chunks
+ * @param {nb.LocationInfo} [location_info]
+ */
+async function update_chunks_on_read(chunks, location_info) {
+    const chunks_to_scrub = [];
+    try {
+        const bucket = chunks[0].bucket;
+        const selected_tier = await map_server.select_tier_for_write(bucket);
+        for (const chunk of chunks) {
+            if (!chunk.tier._id || !_.isEqual(chunk.tier._id, selected_tier._id)) {
```

Comment:
how can a chunk not have tier._id? is that a valid state?

Reply:
if a tier is deleted. Doesn't suppose to happen right now - as we don't allow tier deletions but if it will happen probably we should re-locate it back on the 1st tier.